### PR TITLE
Update the `odo catalog search component` output

### PR DESCRIPTION
**Pull request by cdrage**
Updates the output to match the output from `odo catalog search
service`.

See below:

```sh
github.com/openshift/odo  update-search-table ✗                                                                                                                                                                                                                                                                                                                       6m ⚑
▶ ./odo catalog search service rails
NAME                       PLANS
rails-pgsql-persistent     default

github.com/openshift/odo  update-search-table ✗                                                                                                                                                                                                                                                                                                                       6m ⚑
▶ ./odo catalog search component r
NAME
perl
ruby

github.com/openshift/odo  update-search-table ✗                                                                                                                                                                                                                                                                                                                       6m ⚑
▶ ./odo catalog search component ""
NAME
dotnet
httpd
nginx
nodejs
perl
php
python
ruby
wildfly
```

### CODE REVIEW

**Comment by dharmit** [comment]
@cdrage I'm not sure what you mean by:

> Updates the output to match the output from `odo catalog search service`.

But I see that searching is possible outside a component directory after building odo from this PR. :+1:

**Comment by cdrage** [comment]
> Add integration test

Alright, will do :+1:

**Comment by cdrage** [comment]
Added @amitkrout

File: tests/integration/servicecatalog/cmd_service_test.go [hunk 1]
```diff
@@ -52,6 +52,25 @@ var _ = Describe("odo service command tests", func() {
...
 	})
 
+	Context("check search functionality", func() {
+
+		It("should pass with searching for part of a service name", func() {
+
+			// We just use "sql" as some catalogs only have postgresql-persistent and
+			// others dh-postgresql-db. So let's just see if there's "any" postgresql to begin
+			// with
+			output := helper.CmdShouldPass("odo", "catalog", "search", "service", "sql")
```
Comment (amitkrout) [inline tests/integration/servicecatalog/cmd_service_test.go]: So search fetches the full list of services where some part of name matches.

File: tests/integration/servicecatalog/cmd_service_test.go [hunk 2]
```diff
@@ -52,6 +52,25 @@ var _ = Describe("odo service command tests", func() {
...
+			Expect(output).To(ContainSubstring("postgresql"))
+		})
+
+		It("should fail searching for something nonexistent", func() {
+
+			output := helper.CmdShoulFail("odo", "catalog", "search", "service", helper.RandString(20))
+			Expect(output).To(ContainSubstring("postgresql"))
```
Comment (amitkrout) [inline tests/integration/servicecatalog/cmd_service_test.go]: If it fails to search then IMO it should error out with a proper message of something like ```Service <service_name> not found in catalog list services```

File: tests/integration/servicecatalog/cmd_service_test.go (hunk 1 above)
Comment (cdrage) [reply tests/integration/servicecatalog/cmd_service_test.go]: Yes that's the point, (see comments)

**Comment by cdrage** [comment]
Changing how error works is outside the scope of the PR @amitkrout as this is UX / UI facing.

I've opened up issue: https://github.com/openshift/odo/issues/2147

@amitkrout

**Comment by girishramnani** [comment]
looks good to me
/lgtm

**Comment by amitkrout** [comment]
> Changing how error works is outside the scope of the PR @amitkrout as this is UX / UI facing.
> 
> I've opened up issue: #2147
> 
> @amitkrout

I believe it is not outside the scope of this pr because the search command gives weird output when it fails.
/hold

**Pull request merged**